starknet/src/library/trie_proof.cairo

```diff
@@ -30,36 +30,37 @@ const EMPTY_TRIE_ROOT_HASH: u256 =
 ///
 /// # Returns
 ///
-/// An `Option<Words64Sequence>` containing the data associated with the verified path, or `None` if
+/// An ` Result<Option<Words64Sequence>>` containing the data associated with the verified path, or `Result<felt252>` if
 /// the proof is invalid or the path does not exist in the Merkle Patricia Tree.
 ///
-/// # Panics
-///
-/// The function may panic in the following cases:
+/// # Error(Result<felt252>)
 ///
+/// The function may returns Result<felt252> in the following cases:
 /// - If the proof is empty and the provided `root_hash` is not the empty trie root hash.
 /// - If a hash mismatch is encountered during the verification process.
 /// - If an invalid node length is encountered during the verification process.
 /// - If the path offset exceeds the length of the provided path.
-///
 pub fn verify_proof(
     path: Words64Sequence, root_hash: Words64Sequence, proof: Span<Words64Sequence>,
-) -> Option<Words64Sequence> {
+) -> Result<Option<Words64Sequence>, felt252> {
     let mut out = Option::None;
     let proof_len = proof.len();
 
     if proof_len == 0 {
-        assert!(
-            root_hash.values == split_u256_to_u64_array(EMPTY_TRIE_ROOT_HASH),
-            "Empty proof must have empty trie root hash"
-        );
-        // return Option::Some(Words64Sequence { values: array![].span(), len_bytes: 0 });
-        return Option::None;
+        if root_hash.values != split_u256_to_u64_array(EMPTY_TRIE_ROOT_HASH) {
+            return Result::Err(
+                'Empty proof must be empty'
+            ); //Empty proof must have empty trie root hash
+        } else {
+            return Result::Ok(out);
+        }
     }
 
     let mut next_hash = Words64Sequence { values: array![].span(), len_bytes: 0 };
     let mut path_offset = 0;
     let mut i = 0_u32;
+    let mut return_string = '';
```

Comment:
instead of this see if you can switch from `while loop` to `loop ` and break result directly like [here](https://github.com/starkware-libs/cairo/blob/04ad5913320d903734cd20507ba79feb94f38f85/corelib/src/starknet/storage_access.cairo#L597)